### PR TITLE
Add back khr_vulkan_enable support

### PR DESCRIPTION
**Pull request by zarik5**
As proposed in #84.

The code is mostly copy-pasted from the tag v0.13.1

Tested on the Oculus Quest 2.

### CODE REVIEW

File: openxr/src/instance.rs [hunk 1]
```diff
@@ -225,11 +225,48 @@ impl Instance {
...
-    /// Get a suitable [`VkPhysicalDevice`] for use with a particular `system`
+    /// Identify the Vulkan instance extensions required by a system
+    ///
+    /// Returns a space-delimited list of Vulkan instance extension names.
     ///
-    /// Call [`Instance::create_vulkan_instance()`] first to get a suitable `VkInstance`.
+    /// **Note:** This method requires `khr_vulkan_enable` extension. Applications should use
+    /// `khr_vulkan_enable2` instead, if possible.
+    #[inline]
+    pub fn vulkan_instance_extensions(&self, system: SystemId) -> Result<String> {
```
Comment (Ralith) [inline openxr/src/instance.rs:235]:
```suggestion
    pub fn vulkan_legacy_instance_extensions(&self, system: SystemId) -> Result<String> {
```

File: openxr/src/instance.rs [hunk 2]
```diff
@@ -225,11 +225,48 @@ impl Instance {
...
+    }
+
+    /// Identify the Vulkan device extensions required by a system
+    ///
+    /// Returns a space-delimited list of Vulkan device extension names.
+    ///
+    /// **Note:** This method requires `khr_vulkan_enable` extension. Applications should use
+    /// `khr_vulkan_enable2` instead, if possible.
+    #[inline]
+    pub fn vulkan_device_extensions(&self, system: SystemId) -> Result<String> {
```
Comment (Ralith) [inline openxr/src/instance.rs:254]:
```suggestion
    pub fn vulkan_legacy_device_extensions(&self, system: SystemId) -> Result<String> {
```

File: openxr/src/instance.rs [hunk 3]
```diff
@@ -225,11 +225,48 @@ impl Instance {
         Ok(Ok(instance))
     }
 
-    /// Get a suitable [`VkPhysicalDevice`] for use with a particular `system`
+    /// Identify the Vulkan instance extensions required by a system
+    ///
+    /// Returns a space-delimited list of Vulkan instance extension names.
     ///
-    /// Call [`Instance::create_vulkan_instance()`] first to get a suitable `VkInstance`.
+    /// **Note:** This method requires `khr_vulkan_enable` extension. Applications should use
```
Comment (Ralith) [inline openxr/src/instance.rs:232]:
```suggestion
    /// **Note:** This method requires the `khr_vulkan_enable` extension. Applications should use
```

File: openxr/src/instance.rs [hunk 4]
```diff
@@ -225,11 +225,48 @@ impl Instance {
...
+                buf,
+            )
+        })
+    }
+
+    /// Identify the Vulkan device extensions required by a system
+    ///
+    /// Returns a space-delimited list of Vulkan device extension names.
+    ///
+    /// **Note:** This method requires `khr_vulkan_enable` extension. Applications should use
```
Comment (Ralith) [inline openxr/src/instance.rs:251]:
```suggestion
    /// **Note:** This method requires the `khr_vulkan_enable` extension. Applications should use
```

**Review by Ralith — CHANGES_REQUESTED** [review]
Wow, that's way smaller a change than I expected. Awesome!

Let's name the legacy methods a little more distinctively to avoid confusion with the new ones.

**Comment by zarik5** [comment]
The PR should be ready for merging.

**Review by Ralith — APPROVED** [review]
Thanks!

**Pull request merged**